plugout_sdl.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * gbsplay is a Gameboy sound player
+ *
+ * 2020 (C) by Christian Garbs <mitch@cgarbs.de>
+ * Licensed under GNU GPL v1 or, at your option, any later version.
+ *
+ * SDL2 sound output plugin
+ */
+
+#include "common.h"
+
+#include <time.h>
+
+#include <SDL2/SDL.h>
+
+#include "plugout.h"
+
+static const int PLAYBACK_MODE = 0;
+static const int UNPAUSE = 0;
+
+static const struct timespec SLEEP_INTERVAL = {
+	.tv_sec = 0,
+	.tv_nsec = 1000
+};
+
+int device;
+SDL_AudioSpec obtained;
+
+static long sdl_open(enum plugout_endian endian, long rate)
+{
+	SDL_AudioSpec desired;
+	if (SDL_Init(SDL_INIT_AUDIO) != 0) {
+		fprintf(stderr, _("Could not init SDL: %s\n"), SDL_GetError());
+		return -1;
+	}
+
+	// TODO: what about endianess?
```

Comment:
From https://wiki.libsdl.org/SDL_AudioSpec:
AUDIO_S16LSB | signed 16-bit samples in little-endian byte order
AUDIO_S16MSB | signed 16-bit samples in big-endian byte order
AUDIO_S16SYS | signed 16-bit samples in native byte orde

Reply:
The endianess, it does nothing :-)

I've added endianess selection, but when I try all three settings on my system, the sounds is never broken, so the setting actually does not change anything in the sound output.

I think that gbsplay changes the endianess in the generated sample data, we tell libSDL about the endianess and libSDL converts it to the host format so the sound is always played correctly.

That's not what I expected, but I think it is the right thing™.

Testing with the PulseAudio backend shows the same behaviour.